service/history/execution/mutable_state_task_generator_test.go

```diff
@@ -667,6 +771,931 @@ func (s *mutableStateTaskGeneratorSuite) TestGetNextDecisionTimeout() {
+				Message: fmt.Sprintf("it could be a bug, cannot get pending activity: %v", event.ID),
+			},
+		},
+		{
+			name: "Error case - getTargetDomainID error",
+			setupMock: func(activityInfo persistence.ActivityInfo) {
+				s.mockMutableState.EXPECT().GetActivityInfo(event.ID).Return(&activityInfo, true).Times(1)
+				s.mockDomainCache.EXPECT().GetDomainID(event.ActivityTaskScheduledEventAttributes.GetDomain()).
+					Return("", errors.New("get-target-domain-id-error")).Times(1)
+			},
+			err: errors.New("get-target-domain-id-error"),
+		},
+	}
+
+	for _, tc := range testCases {
+		s.T().Run(tc.name, func(t *testing.T) {
+			tc.setupMock(activityInfo)
+
+			err := s.taskGenerator.GenerateActivityTransferTasks(event)
+
+			if tc.err != nil {
+				s.Error(err)
+				s.Equal(tc.err, err)
+			} else {
+				s.NoError(err)
+			}
+		})
+	}
+}
+
+func (s *mutableStateTaskGeneratorSuite) TestGenerateActivityRetryTasks() {
+	activityScheduleID := int64(123)
+
+	testCases := []struct {
+		name      string
+		setupMock func()
+		err       error
+	}{
+		{
+			name: "Success case",
+			setupMock: func() {
+				ai := &persistence.ActivityInfo{
+					Version:       constants.TestVersion,
+					ScheduledTime: time.Now(),
+					ScheduleID:    activityScheduleID,
+					Attempt:       1,
+				}
+				s.mockMutableState.EXPECT().GetActivityInfo(activityScheduleID).Return(ai, true).Times(1)
+				s.mockMutableState.EXPECT().AddTimerTasks(&persistence.ActivityRetryTimerTask{
+					TaskData: persistence.TaskData{
+						Version:             ai.Version,
+						VisibilityTimestamp: ai.ScheduledTime,
+					},
+					EventID: ai.ScheduleID,
+					Attempt: ai.Attempt,
+				}).Times(1)
+			},
+		},
+		{
+			name: "Error case - GetActivityInfo error",
+			setupMock: func() {
+				s.mockMutableState.EXPECT().GetActivityInfo(activityScheduleID).Return(nil, false).Times(1)
+			},
+			err: &types.InternalServiceError{
+				Message: fmt.Sprintf("it could be a bug, cannot get pending activity: %v", activityScheduleID),
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		s.T().Run(tc.name, func(t *testing.T) {
+			tc.setupMock()
+
+			err := s.taskGenerator.GenerateActivityRetryTasks(activityScheduleID)
+
+			if tc.err != nil {
+				s.Error(err)
+				s.Equal(tc.err, err)
+			} else {
+				s.NoError(err)
+			}
+		})
+
+	}
+}
+
+func (s *mutableStateTaskGeneratorSuite) TestGenerateChildWorkflowTasks() {
+	eventID := int64(123)
+
+	childWorkflowInfo := persistence.ChildExecutionInfo{
+		Version:           constants.TestVersion,
+		InitiatedID:       123,
+		StartedWorkflowID: constants.TestWorkflowID,
+	}
+
+	testCases := []struct {
+		name       string
+		setupMock  func(childWorkflowInfo persistence.ChildExecutionInfo)
+		domainName string
+		err        error
+	}{
+		{
+			name: "Success case - targetDomainID is not empty and isCrossClusterTask is false",
+			setupMock: func(childWorkflowInfo persistence.ChildExecutionInfo) {
+				childWorkflowInfo.DomainID = constants.TestDomainID
```

Comment:
same as before, this is a bit of a problem for subsequent tests, if it's at all possible to just return a new childWorkflowInfo for each unit in the table test it'll be less confusing for debugging should there need to be more.